src/unix.rs

```diff
@@ -596,6 +616,19 @@ pub fn set_thread_priority_and_policy(
                     e => Err(Error::OS(e)),
                 }
             } else {
+                //VxWorks does not have set priority function
+                #[cfg(target_os = "vxworks")]
+                unsafe fn setpriority(
+                    _which: u32,
+                    _who: u32,
+                    _priority: libc::c_int,
+                ) -> libc::c_int {
+                    panic!("Set priority function not supported by vxworks");
+                }
+
+                #[cfg(not(target_os = "vxworks"))]
+                use libc::setpriority;
+
```

Comment:
I agree with the bot, I think it is better to allow errors and let the user handle them, even if the user is us.

Reply:
But set priority does not return an error type. Plus due to the cfg in if condition, control would never reach this call. But because the cfg is added as a boolean to if condition, I get build errors when compiling this due to setpriority not available in vxworks.

Reply:
@coderabbitai, the changes have been made.

Reply:
@coderabbitai Suggested changes have been committed.